src/main.js

```diff
@@ -112,19 +113,26 @@ function favoriteIdea(event) {
   if (event.target.id === "favoriteButton") {
     var ideaId = event.target.closest("article").id;
 
-    ideas[ideaId].updateIdea();
-    //ideas[ideaId].saveToStorage(ideas[ideaId]);
+  for (var i = 0; i < ideas.length; i++) {
+    if (parseInt(ideaId) === ideas[i].id) {
+      ideas[i].updateIdea();
+      ideas[i].saveToStorage();
+    }
+  }
     render(ideas);
   }
 }
 
 function deleteIdea(event) {
   if (event.target.id === "deleteButton") {
     var ideaId = event.target.closest("article").id;
-    
-    ideas[ideaId].deleteFromStorage();
-    ideas.splice(ideaId, 1); 
-    
+
+    for (var i = 0; i < ideas.length; i++) {
```

Comment:
This adds a double loop to our app, which isn't the worst thing, but isn't needed.